src/app/exam/exam-detail/exam-detail.component.ts

```diff
@@ -0,0 +1,55 @@
+
+import { Component, OnInit } from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {AuthenticationBasicService} from '../../login-basic/authentication-basic.service';
+import {DomSanitizer, SafeResourceUrl} from '@angular/platform-browser';
+import {Sort} from '@lagoshny/ngx-hal-client';
+
+import {User} from '../../login-basic/user';
+import { Exam } from '../exam';
+import { ExamService } from '../exam.service';
+import { Question } from '../../question/question';
+import { QuestionService } from '../../question/question.service';
+
+
+@Component({
+  selector: 'app-exam-detail',
+  templateUrl: './exam-detail.component.html'
+})
+export class ExamDetailComponent implements OnInit {
+
+  public exam: Exam = new Exam();
+  public questions: Question[] = [];
+
+  GenuineUrl: SafeResourceUrl;
+  private sorting: Sort[] = [{ path: 'statement', order: 'ASC' }];
+
+  constructor(
+    public route: ActivatedRoute,
+    public examService: ExamService,
+    private questionService: QuestionService,
+    public authenticationService: AuthenticationBasicService) {
+  }
+
+  ngOnInit() {
+    console.log("exam detail init");
+
+    const id = this.route.snapshot.paramMap.get('id');
+
+    this.examService.get(id).subscribe(
+      exam => {
+        this.exam = exam;
```

Comment:
The questions for an exam can be retrieved following the corresponding _link. This is facilitated by the ngx_hal_client dependency as I showed in the last recording. Something like:

```
this.exam.getRelationArray(Question, 'questions').subscribe(questions => exam.questions = questions))
```